rbd-target-api.py

```diff
@@ -2816,13 +2817,46 @@ def run(self):
                     config.refresh()
 
 
+def get_ssl_files_from_mon():
+    client_name = settings.config.cluster_client_name
+    temp_files = []
+    with rados.Rados(conffile=settings.config.cephconf,
+                     name=client_name) as cluster:
+        cmd = {"prefix": "mon config-key get",
+               "key": "iscsi/{}/iscsi-gateway.crt".format(client_name)}
+        ret, crt_data, outs = cluster.mon_command(json.dumps(cmd), b'')
+        if not ret:
+            return temp_files
+
+        cmd["key"] = "iscsi/{}/iscsi-gateway.key".format(client_name)
+        ret, key_data, outs = cluster.mon_command(json.dumps(cmd), b'')
+        if not ret:
+            return temp_files
+    for data in crt_data, key_data:
+        # NOTE: Annoyingly SSLContext.load_cert_chain can only take
+        # paths to files and not file like objects.. yet. So we need to
+        # create tempfiles for the SSL context to read. Once
+        # https://bugs.python.org/issue16487 is resolved, we should be able
+        # to simply use file-like objects and makes this much nicer.
+
+        tmp_f = tempfile.TemporaryFile()
```

Comment:
Nit: you would need to keep this `tmp_f` in-scope so that it doesn't get garbage collected. Otherwise there is a potential race of the temp files being deleted prior to use.

Reply:
Ahh yeah, very good spot. I originally inlined this in the `get_ssl_context` but started getting a bit big so pulled it out, yes this could very much be an issue!